src/components/GenericErrorView.js

```diff
@@ -15,16 +21,40 @@ class GenericErrorView extends PureComponent<{
   outerError?: ?Error,
   withDescription: boolean,
   withIcon: boolean,
+  hasExportLogButton?: boolean,
 }> {
   static defaultProps = {
     withDescription: true,
     withIcon: true,
   };
 
+  onExport = async () => {
+    const logs = logReport.getLogs();
+    const message = logs.map(log => JSON.stringify(log)).join("\n");
+    const base64 = Buffer.from(message).toString("base64");
+
+    const options = {
+      failOnCancel: false,
+      saveToFiles: true,
+      type: "application/json",
+      filename: "logs",
+      url: `data:application/json;base64,${base64}`,
+    };
+
+    try {
+      await Share.open(options);
+    } catch (err) {
+      if (err.error.code !== "ECANCELLED500") {
+        logger.critical(err);
+      }
+    }
+  };
   render() {
     const { error, outerError, withDescription, withIcon } = this.props;
     const titleError = outerError || error;
     const subtitleError = outerError ? error : null;
+    const hasExportLogButton = this.props.hasExportLogButton !== false; // hasExportLogButton is true by default
+
```

Comment:
Currently, I put hasExportLogButton = true by default. It means by default, an error message has this "save logs" button.
but I check the existing errors, it seems that not all the errors need this button. because some errors are only a warning message to users. We don't want them to save log and send to Ledger support. I leave the possibility to hide the button with this variable. 
I don't know whether I should config all the old errors so that we remove this button for certain errors? if yes, I need someone else to help me with that

Reply:
I've talked with @cthepot-ledger offline. We decided to refer to the current logic of ledger-live-desktop app.
I found that in LLD, the "save logs" button always appears.

refer to:
https://github.com/LedgerHQ/ledger-live-desktop/blob/766fdf662b78d1d9824367687a5ffe725feef730/src/renderer/components/DeviceAction/index.js#L177
and
https://github.com/LedgerHQ/ledger-live-desktop/blob/develop/src/renderer/components/DeviceAction/rendering.js#L344

Screen shot from LLD:
<img width="1020" alt="Screenshot 2021-04-12 at 15 08 40" src="https://user-images.githubusercontent.com/71653044/114518660-df68eb80-9c3f-11eb-8c28-938265e8a746.png">

@gre can you confirm that?

So, I assume that the my current work has the same behaviour of LLD and we can merge this PR after review.

Reply:
yes it always appears for unhandled errors like this one. so yeah to me we can be iso.